docs/modules/ROOT/pages/references/architecture/control-api-org.adoc

```diff
@@ -63,24 +63,82 @@ metadata:
 
 |===
 
-== Resource filter
+=== Resource filter
 
 The virtual resource is a filtered view of `Namespaces`.
 The filter uses the following heuristic:
 
 * API version: `v1`
 * Kind: `Namespace`
 * Label: `appuio.io/resource-type=organization`
-* Subject is bound to one of the defined `ClusterRole` resources.
 
-== RBAC and Cluster roles
+== Access Control
 
-These are `ClusterRole` resources which are bound to a subject by a namespaced `RoleBinding`:
+Access control for `Organizations` uses standard Kubernetes RBAC rules, but `Organizations` are represented by two different resources.
+
+. The `organizations.organizations.appuio.io` resource is the cluster-scoped resource we actually access.
+The access control for this resource is done by the Kubernetes API server.
+RBAC works the same as for any other cluster-scoped resources and can be configured through `ClusterRoles` and `ClusterRoleBindings`.
+
+. The `organizations.rbac.appuio.io` resource is a custom namespace-scoped resource known only to the `Organization` API server, which is only used for access control.
+Being namespace-scoped, permissions can also be configured through `Roles` and `RoleBindings`.
+Similarly to `Namespaces`, permissions configured by a `Role` in namespace `foo` only affects `Organization` `foo`.
+
+A user can only perform an action on an `Organization`, if the action is allowed on both of these resources.
+The `list` and `watch` verbs are special cases and the API server will only return resources that the user also has permission to `get`
+
+In practice this means, we will allow any user to perform any action on `organizations.organizations.appuio.io` and manage access to `Organizations` through `organizations.rbac.appuio.io`.
+
+[NOTE]
+====
+This trick of using `organizations.rbac.appuio.io` allows us to handle and especially delegate access to `Organizations` on a namespace level.
+So we can allow users to configure access without having to expose `ClusterRoleBindings`.
+====
+
+=== Generated RBAC
+
+By default there are two `ClusterRoles` that configure access for organization members
 
 `appuio-organization-viewer`:: View (read only) access to an organization
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: appuio-organization-viewer
+rules:
+- apiGroups: ["rbac.appuio.io"]
+  resources: ["organizations"]
+  verbs: ["get"]
+...
+# Get and list permission for other resources
+----
+
 `appuio-organization-admin`:: Admin (read / write) access to an organization
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: appuio-organization-admin
+rules:
+- apiGroups: ["rbac.appuio.io"]
```

Comment:
I'm not quite sure we actually need a custom group `rbac.appuio.io`. can the custom API server not just verify (in addition to standard Kubernetes RBAC) to see if there's additional permissions in a namespace-scoped rolebinding?

Maybe a namespace-scoped example is needed to better understand why this special group is needed? Because I'm guessing this is done to allow users of other organizations to access it?

Reply:
Theoretically yes we probably could add additional checks in our API server to see whether there is a rolebinding to explicitly allow accessing an organization. In practice I expect this to get hairy.

There is no way (that I found) to disable standard Kubernetes RBAC for organization. This means we will have to allow any action on organizations for everyone on a cluster scope. Otherwise our API server never gets the request. So we would only allow actions on organizations if the action is allowed on a cluster and namespace level. I see two issues with that:

* This is IMO not really more intuitive that adding another custom group and this will be annoying for cluster-admins as we do now not have any option to enable cluster wide access to organizations
* We need to implement parsing roles/rolebindings ourselves. Currently we can basically use `kubectl auth can-i` to check if a user is allowed to perform actions on a `organizations.rbac.appuio.io` resource. Re-implementing these security checks is not something I want to do as it's error prone and errors have security implications.

I will try to write something that better explains the use of this new group.

Reply:
Thanks for the clarification. I think especially the second bullet should actually make it into the page, as this is a quite important aspect IMO.